view/frontend/web/js/view/cart/product-protection-simple-offer.js

```diff
@@ -38,7 +38,14 @@ define(['cartUtils', 'extendSdk', 'ExtendMagento'], function (cartUtils, Extend,
         listPrice,
         offerId,
         quantity: quantity ?? getProductQuantity(cartItems, product),
-      }).then(cartUtils.refreshMiniCart)
+      }).then(function () {
+        // The underlying code in the refreshMiniCart function forces a cart
+        // invalidation that effects more than just the mini cart. We rely on this
+        // to happen before refreshing so that we never get a stale state where the
+        // refreshed page cart is missing the virtual product and thus tries to show offers.
+        cartUtils.refreshMiniCart()
+        window.location.reload()
```

Comment:
Are we sure that this is not causing `simple offer button` getting rendered that we saw during huddle?

Reply:
Yeah I tested both the subscribe and this method and they're equivalent.

The reason we saw that happen in the huddle is because I replaced the entire `then` including the `refreshMiniCart` function. It turns out that the `refreshMiniCart` function works by invalidating the entire cart which forces a reload. As long as this refreshMiniCart function stays, this will work which is why I added that comment.